### PR TITLE
Parallelize certain AQL write operations

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Extend the existing `parallelize-gather` AQL optimizer rule so it supports certain write queries (REMOVE, UPDATE, REPLACE that do not use a DistributeNode - all other restrictions for the `parallelize-gather` optimizer rule still apply!). 

- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)

### Testing & Verification

This change is a trivial rework / code cleanup without any test coverage.

https://172.16.10.101/view/PR/job/arangodb-matrix-pr/7322/

### CODE REVIEW

File: arangod/Aql/ClusterNodes.cpp [hunk 1]
```diff
@@ -562,20 +569,15 @@ struct ParallelizableFinder final : public WalkerWorker<ExecutionNode> {
...
-        && 
-        (node->getType() != ExecutionNode::REMOVE &&
-         node->getType() != ExecutionNode::REPLACE && 
-         node->getType() != ExecutionNode::UPDATE)) {
-         */
-      // REMOVEs and REPLACEs are actually parallelizable, as they are completely independent
-      // from each other on different shards
-      _isParallelizable = false;
-      return true;  // true to abort the whole walking process
+      if (!_vocbase.server().getFeature<OptimizerRulesFeature>().parallelizeGatherWrites() ||
```
Comment (graetzer) [inline arangod/Aql/ClusterNodes.cpp]: can you not get the server through the engine ?

**Comment by jsteemann** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/7422/

**Comment by jsteemann** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/7433/

**Comment by jsteemann** [comment]
Tests blue

**Comment by jsteemann** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/7447/

**Comment by KVS85** [comment]
https://172.16.10.101/job/arangodb-matrix-pr-linux/9232/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**